protostar/cheatable_starknet/controllers/contracts.py

```diff
@@ -55,11 +68,22 @@ class ConstructorInvocationException(ContractsCheaterException):
     pass
 
 
+class OperationNotSupportedForCairo1Exception(ProtostarException):
+    def __init__(self, message: str = "Operation not supported for Cairo1 contracts."):
+        super().__init__(message=message)
```

Comment:
I would raise CheatcodeException there instead, is there any reason why we need to create a new error type here?

Reply:
The previous suggestion would be to raise a specific error here so it can be handled by the cheatcodes. But I guess it doesn't make sense since it cannot be reasonably handled.